### PR TITLE
Rename internals xdai->gnosis and allow --network=gnosis in cardpay CLI

### PR DESCRIPTION
**Pull request by lukemelia**

### CODE REVIEW

File: packages/cardpay-sdk/sdk/constants.ts [hunk 1]
```diff
@@ -89,7 +89,8 @@ const constants: {
   sokol: SOKOL,
   kovan: KOVAN,
   mainnet: MAINNET,
-  xdai: XDAI,
+  gnosis: GNOSIS,
+  xdai: GNOSIS,
 });
 
 export const networks: { [networkId: number]: string } = Object.freeze({
```
Comment (backspace) [inline packages/cardpay-sdk/sdk/constants.ts:96]: a few lines down from here is one other place that needs updating; if you visit [the `web-client` productionpreview deployment] there‘s this error in the console:

```
Error: chainId is required to set custom session storage for parallel connections in WalletConnect
```

**Review by backspace — CHANGES_REQUESTED** [review]
This mostly looks good to me apart from the web breakage but also this breaks existing payment links, you can see that on [this preview deployment](https://cs-4120-cardpay-cli-should-accept-gnosis-as-a.ssr-web-preview.cardstack.com/pay/xdai/0xf9c0E2B59824f33656CC5A94423FcF62892dad60?amount=0.5&currency=USD). It works if you change `xdai` in the URL to `gnosis`, but I always prefer to keep old links working, so maybe this means forwarding in `ssr-web`? Or maybe the `networkIds` change would fix this too, I’m unsure.

File: packages/cardpay-sdk/sdk/constants.ts (hunk 1 above)
Comment (lukemelia) [reply packages/cardpay-sdk/sdk/constants.ts:96]: Actually, I think ssr-web is good but web-client still has issues.

Comment (lukemelia) [reply packages/cardpay-sdk/sdk/constants.ts:96]: @backspace Now this should be all good.

**Review by backspace — APPROVED** [review]
I’m glad the unraveling ended eventually!

**Review by jurgenwerk — APPROVED** [review]
👏

**Pull request merged**